rune/libenclave/internal/runtime/pal/api_linux_v3.go

```diff
@@ -0,0 +1,58 @@
+package enclave_runtime_pal // import "github.com/opencontainers/runc/libenclave/internal/runtime/pal"
+
+/*
+#include <stdlib.h>
+#include <errno.h>
+
+static int pal_get_reportV3(void *sym, void *target_info, int target_info_len,
+			void *data, int data_len, void *report, int* report_len)
+{
```

Comment:
You are assuming sgx sdk get report api as the direct underlaying function but it is not true. Data is enclave-specific data structure and it should be filled inside enclave. So we don't need to pass data parameter. Again, you cannot assume get report api from sgx sdk is the only underlaying function.